org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/commands/ProjectCommand.java

```diff
@@ -270,13 +270,17 @@ public static IJavaProject getJavaProjectFromUri(String uri) throws CoreExceptio
 		}
 
 		// For multi-module scenario
-		Arrays.sort(containers, (Comparator<IContainer>) (IContainer a, IContainer b) -> {
-			return a.getFullPath().toPortableString().length() - b.getFullPath().toPortableString().length();
-		});
+		IProject[] projects = Arrays.stream(containers).map((container) -> {
+			return container.getProject();
+		}).filter(Objects::nonNull)
+		.sorted((IProject a, IProject b) -> {
+			// inner comes first
+			return b.getFullPath().toPortableString().length() - a.getFullPath().toPortableString().length();
```

Comment:
shouldn't you count the path segments instead?

Reply:
How about `b.getLocation().toPortableString().length() - a.getLocation().toPortableString().length();`?

Use `getLocation()` to get the absolute path on FS.

Reply:
Hmm... I just realized that there will be another corner case for relative-path module like this:

```
<modules>      
    <module>../subproject1<module>
    <module>../subproject2<module>
</modules>
```

this will fail both length or segment count.

Maybe the best way to handle this is get all Java projects in workspace and iterate over the Java projects. Find the one whose path is prefix of the input and has the longest length?